cluster/store/store.go

```diff
@@ -255,25 +258,23 @@ func (st *Store) Open(ctx context.Context) (err error) {
 		return fmt.Errorf("raft.NewRaft %v %w", st.transport.LocalAddr(), err)
 	}
 
-	snapIndex := snapshotIndex(st.snapshotStore)
-	if st.lastAppliedIndexOnStart.Load() == 0 && snapIndex == 0 {
-		// if empty node report ready
-		st.dbLoaded.Store(true)
-	}
-
 	st.lastAppliedIndex.Store(st.raft.AppliedIndex())
 
 	st.log.WithFields(logrus.Fields{
 		"raft_applied_index":           st.raft.AppliedIndex(),
 		"raft_last_index":              st.raft.LastIndex(),
 		"last_store_log_applied_index": st.lastAppliedIndexOnStart.Load(),
 		"last_store_applied_index":     st.lastAppliedIndex.Load(),
-		"last_snapshot_index":          snapIndex,
+		"last_snapshot_index":          snapshotIndex(st.snapshotStore),
 	}).Info("raft node constructed")
 
 	// There's no hard limit on the migration, so it should take as long as necessary.
 	// However, we believe that 1 day should be more than sufficient.
-	go st.onLeaderFound(time.Hour * 24)
+	// if empty node report ready and apply migration
+	if st.IsEmptyNode() {
```

Comment:
We already have https://github.com/weaviate/weaviate/blob/main/cluster/store/store.go#L365, do you think it's insufficient ?  Also I would move this check down to the migration "hook" or code path and leave the `onLeaderFound` always trigger. We might extend it in the future and use it for other topics apart from migration.

Reply:
aah , gotcha didn't see `st.db.Schema.len() == 0` i think it's sufficient. i just didn't see this part . I can close this PR then and we improve later 👍🏼

Reply:
what we may need to verify given the migration runs on go routine if the schema is there but not just loaded in mem.